operator/metrics/metrics.go

```diff
@@ -169,6 +169,7 @@ func registerMetrics() []prometheus.Collector {
 		Namespace: Namespace,
 		Name:      "number_of_ceps_per_ces",
 		Help:      "The number of CEPs batched in a CES",
+		Buckets:   prometheus.LinearBuckets(0, 10, operatorOption.Config.CESMaxCEPsInCES/10),
```

Comment:
I worry about directly tying this into the configuration since adjusting the configuration value will result in buckets changing and histograms no longer being compatible with their previous labels. It might be nice if this was configurable separately, but I imagine we can pick reasonable buckets.

I imagine the following buckets would be fairly reasonable and would support smaller and larger environments:
```
Buckets: []float64{10, 25, 50, 100, 200, 500, 1000},
```

Reply:
Constant width of the buckets is suggested by comment in [L90-95](https://github.com/cilium/cilium/blob/e25885f1e7db74b84b7d3ebca96642cd64b8f40f/operator/metrics/metrics.go#L90). I think that with constant width it doesn't make sense to have hand-picked buckets as each one above `CESMaxCEPsInCES` should be empty.

Labels should match for different `CESMaxCEPsInCES` values:
```
CESMaxCEPsInCES = 20: le=0, le=10, le=20
CESMaxCEPsInCES = 50: le=0, le=10, le=20, le=30, le=40, le=50
```

The only problem I could foresee is `CESMaxCEPsInCES` changing without updating the buckets. Can it be changed in the runtime?

Another thing to consider, should we remove the lowest bucket and start with `le=10`?

Reply:
Fundamentally my issue is using one configuration value for determining histogram buckets, because changing histogram buckets can be disruptive to existing Prometheus queries, hence the suggestion to "hand-pick/hardcode" them.

You're correct though that we shouldn't see values above the configured maximum though, so that does make it a bit tricky with hand-picked values. I also agree that a constant width of 10 won't work for higher values. 

Regarding the lowest bucket, having a bucket for `[0,1]` may make sense, due to many apps potentially running 1 replica services, and it may be useful to distinguish that from the `[0,10]` bucket.

Overall, this is making me think we may just want to expose this as a configuration option with some reasonable default sizes. I don't believe we expose configuring bucket sizes for any existing metrics, but it's been a topic previously, since picking good buckets that work for both small and large environments can be very difficult.

What are your thoughts?

Reply:
I think that making buckets configurable is a bit extreme. It would be great to have sane defaults and the default values definitely don't work well for this type of data.

I changed these buckets to what you proposed and `le=1`